src/rest_vol.c

```diff
@@ -1257,10 +1256,22 @@ H5_rest_curl_read_data_callback(char *buffer, size_t size, size_t nmemb, void *i
     size_t       max_buf_size = size * nmemb;
     size_t       data_size = 0;
 
+    /* If all bytes sent, indicate transfer is finished */
+    if (uinfo->bytes_sent >= uinfo->buffer_size) {
+        return 0;
+    }
```

Comment:
This block should be moved inside the `if (inptr)` block below so that we don't attempt to dereference `uinfo` in case `inptr` is NULL.